File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/SignBreakEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;
import top.likoslupus.cellulosesz.api.teleport.CellLocation;

import java.util.List;
import java.util.Objects;

public final class SignBreakEvent extends AbstractCancellableEvent {

    private final CellPlayer player;
    private final CellLocation location;
    private final List<String> frontLines;
    private final List<String> backLines;

    public SignBreakEvent(
            CellPlayer player,
            CellLocation location,
            List<String> frontLines,
            List<String> backLines
    ) {
        this.player = Objects.requireNonNull(player, "player");
        this.location = Objects.requireNonNull(location, "location");
        this.frontLines = List.copyOf(Objects.requireNonNull(frontLines, "frontLines"));
        this.backLines = List.copyOf(Objects.requireNonNull(backLines, "backLines"));
    }

    public CellPlayer player() {
        return player;
    }

    public CellLocation location() {
        return location;
    }

    public List<String> frontLines() {
        return frontLines;
    }

    public List<String> backLines() {
        return backLines;
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/SignBreakEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import top.likoslupus.cellulosesz.api.teleport.CellLocation

public class SignBreakEvent(
    private val player: CellPlayer,
    private val location: CellLocation,
    private val frontLines: List<String>,
    private val backLines: List<String>
) : AbstractCancellableEvent() {

    public fun player(): CellPlayer = player

    public fun location(): CellLocation = location

    public fun frontLines(): List<String> = frontLines

    public fun backLines(): List<String> = backLines

}
